Objects/descrobject.c

```diff
@@ -633,11 +633,40 @@ descr_reduce(PyDescrObject *descr, PyObject *Py_UNUSED(ignored))
                          PyDescr_TYPE(descr), PyDescr_NAME(descr));
 }
 
+
+static PyObject*
+classmethoddescr_reduce(PyDescrObject* descr, PyObject* Py_UNUSED(ignored))
+{
+    /* Ideally, we would want to use a different callable than eval in
+       order to get the descriptor. However, we need to ensure the pickled
+       object will not cause errors upon unpickling in older versions. */
+    PyObject* evalFunctionName = PyUnicode_FromString("eval");
+    if (evalFunctionName == NULL) {
+        return NULL;
+    }
+    PyObject* eval = _PyEval_GetBuiltin(evalFunctionName);
+    Py_DECREF(evalFunctionName);
```

Comment:
I recommend using `_Py_ID()` here. You need to include `pycore_runtime.h` too for that (but it's probably already included).
```suggestion
    PyObject* eval = _PyEval_GetBuiltin(&_Py_ID(eval));
```
Make sure to regen the frozen files after using the Py_ID (this means run `make regen-all` or `build.bat --regen` then `build.bat` again)

Reply:
Ohhh I was wondering about that. The existing descriptor `__reduce__` implementation has used `_Py_ID ` for `getattr` and I was wondering why I couldn't do the same for `eval`. 

Awesome! :D

Updated the PR accordingly :)